### PR TITLE
fix: font size text being displayed out of range.

### PR DESCRIPTION
**Pull request by ZakAnun**
### Feature Preview
Fix [issue 3749](https://github.com/AppFlowy-IO/AppFlowy/issues/3749). By following changes:
1. Increase PopupMenuButton maximum width to adapt to sub widget width display.
2. Use SegmentedButton to replace ToggleButton for better display.

image
image
image

fix [#3749 ](https://github.com/AppFlowy-IO/AppFlowy/issues/3749)

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/document/presentation/more/font_size_switcher.dart [hunk 1]
```diff
@@ -21,13 +22,19 @@ class _FontSizeSwitcherState extends State<FontSizeSwitcher> {
     (LocaleKeys.moreAction_medium.tr(), 18.0, true),
     (LocaleKeys.moreAction_large.tr(), 22.0, false),
   ];
+  Set<(String, double, bool)> _selection = <(String, double, bool)>{};
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/more/font_size_switcher.dart]:
```dart
typedef _FontSizeSelection = (String, double, bool);
```

**Comment by LucasXu0** [comment]
@ZakAnun I did some refactors.

- Replace `(String, double, bool)` with `typedef _FontSizeSelection = (String, double, bool);`.
- Change the type of `_selection` from `Set<...>` to `_FontSizeSelection` since only one selection will be enabled.
- Replace the popover menu with AppFlowyPopover to enable automatic size adjustment.

**After**
Screenshot 2023-10-24 at 13 15 14

**Comment by ZakAnun** [comment]
> ## [Codecov](https://app.codecov.io/gh/AppFlowy-IO/AppFlowy/pull/3766?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=AppFlowy-IO) Report
> Attention: `23 lines` in your changes are missing coverage. Please review.
> 
> > Comparison is base [(`25a98cd`)](https://app.codecov.io/gh/AppFlowy-IO/AppFlowy/commit/25a98cda81c13eeb64b75bd2c4b21a39f826d7f5?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=AppFlowy-IO) 64.99% compared to head [(`6ca762a`)](https://app.codecov.io/gh/AppFlowy-IO/AppFlowy/pull/3766?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=AppFlowy-IO) 11.71%.
> > Report is 2 commits behind head on main.
> 
> Additional details and impacted files
> [☔ View full report in Codecov by Sentry](https://app.codecov.io/gh/AppFlowy-IO/AppFlowy/pull/3766?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=AppFlowy-IO). 📢 Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=AppFlowy-IO).

@LucasXu0 @MayurSMahajan @Xazin Hello, I'm not sure if I need to make any other adjustments to this result.

**Review by LucasXu0 — APPROVED** [review]

**Comment by LucasXu0** [comment]
> > ## [Codecov](https://app.codecov.io/gh/AppFlowy-IO/AppFlowy/pull/3766?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=AppFlowy-IO) Report
> > Attention: `23 lines` in your changes are missing coverage. Please review.
> > > Comparison is base [(`25a98cd`)](https://app.codecov.io/gh/AppFlowy-IO/AppFlowy/commit/25a98cda81c13eeb64b75bd2c4b21a39f826d7f5?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=AppFlowy-IO) 64.99% compared to head [(`6ca762a`)](https://app.codecov.io/gh/AppFlowy-IO/AppFlowy/pull/3766?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=AppFlowy-IO) 11.71%.
> > > Report is 2 commits behind head on main.
> > 
> > 
> > Additional details and impacted files
> > [☔ View full report in Codecov by Sentry](https://app.codecov.io/gh/AppFlowy-IO/AppFlowy/pull/3766?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=AppFlowy-IO). 📢 Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=AppFlowy-IO).
> 
> @LucasXu0 @MayurSMahajan @Xazin Hello, I'm not sure if I need to make any other adjustments to this result.

No. Thanks for your contribution.

**Pull request merged**